asv_bench/benchmarks/attrs_caching.py

```diff
@@ -15,7 +15,7 @@ def setup(self):
         self.cur_index = self.df.index
 
     def time_get_index(self):
-        self.foo = self.df.index
+        self.foo_index = self.df.index
```

Comment:
This is a benchmark for timing get index, so probably clearer to just not assign the result to `self` instead of using an arbitrary name. Might also give more accurate results since not also including time of setting attribute on `self`

Could do something like `self.df.index` or `_index = self.df.index` if the former gives some kind of warning about assignment that does nothing.

Reply:
thx, I didn't notice any warnings about the self-assignment.

Do you have any suggestions for the Foo class in the deprecate_nonkeyword_arguments test?

Reply:
thx, please see below